lib/ui/window/platform_message_response_dart.cc

```diff
@@ -63,11 +68,35 @@ PlatformMessageResponseDart::~PlatformMessageResponseDart() {
 }
 
 void PlatformMessageResponseDart::Complete(std::unique_ptr<fml::Mapping> data) {
-  PostCompletion(std::move(callback_), ui_task_runner_, &is_complete_, channel_,
-                 [data = std::move(data)] {
-                   return tonic::DartByteData::Create(data->GetMapping(),
-                                                      data->GetSize());
-                 });
+  PostCompletion(
+      std::move(callback_), ui_task_runner_, &is_complete_, channel_,
+      [data = std::move(data)]() mutable {
+        Dart_Handle byte_buffer;
+        intptr_t size = data->GetSize();
+        if (data->GetSize() > tonic::DartByteData::kExternalSizeThreshold) {
+          const void* mapping = data->GetMapping();
+          byte_buffer = Dart_NewUnmodifiableExternalTypedDataWithFinalizer(
+              /*type=*/Dart_TypedData_kByteData,
+              /*data=*/mapping,
+              /*length=*/size,
+              /*peer=*/data.release(),
+              /*external_allocation_size=*/size,
+              /*callback=*/MappingFinalizer);
+        } else {
+          Dart_Handle mutable_byte_buffer =
```

Comment:
Correct me if I'mr eading this wrong, but wont' this just call `Dart_NewExternalTypedDataWithFinalizer` just to pass it back through to Dart land and get back an unmodifiable view?

Elsewhere we use this external threshold to just determine whether we should copy the data into the dart heap or not (i.e. whether to use ExternalTypedData or not).

Reply:
Nope, if it is large it will use `Dart_NewUnmodifiableExternalTypedDataWithFinalizer` and return the result.  If it is small it will use `Dart_NewTypedData`, memcpy into it, then send it to be wrapped.  There isn't a way to do that all in one step.  I found in local testing that using `Dart_NewTypedData` plus memcpy was faster for small buffers, and this matched the logic inside of DartByteData already.

Reply:
Ahhh I see, ok.

Maybe as a nitpick it'd make sense to update `DartByteData::Create` to take a flag about whether it's unmodifiable.

I still think it would probably be good to have a test that exercises this and makes sure the typed data can't be written to (and that we don't, for example, segfault if someone tries to write to it).

Reply:
Okay, will do.  Give me a minute.  Looks like there is a general lack of testing on that class anyways so doesn't hurt to backfill some coverage in the engine.

Reply:
Done, added test.